RecoTauTag/RecoTau/python/tools/runTauIdMVA.py

```diff
@@ -172,7 +172,7 @@ def runTauID(self):
                         variable = self.cms.string("pt"),
                     )
                 ),
-                workingPoints = cms.vstring(
+                workingPoints = self.cms.vstring(
```

Comment:
was it broken or is the "self." superfluous in this case?
(same question for many other cases).

Reply:
The "self." is mandatory here and other places as the `TauIDEmbedder` python class is designed such that the global `cms` is a member of the class. The "self." was not added/was removed when tool have been updated to new tau discriminant dataformat with its configuration.

Reply:
what was the rationale for this?
it seems rather burdensome to have to prefix almost everything with "self"

Reply:
@kandrosov, do you remember why the global `cms` (holding definitions of classes used for the python CMSSW configuration) is a member of the `TauIDEmbedder` class?

Regardless, the main intention of the PR is to quickly fix the bug rather than improving the tool.

Reply:
well, if the missing addition of ```self.``` was the leading origin for the bug
I'd rather we do not keep the error-prone implementation.
It's perhaps OK to postpone the update that does not require the ```self.``` for a later PR; however, I think that a unit test is needed now with this PR.

Reply:
It is now added to unit tests. 
I hope we will hear soon about a rationale of usage "self." from Konstantin who (as far as I remember) introduced it.

Reply:
After some thinking, I do not see any rationale of the current design and I agree it is error probe. Therefore, I will commit changes with direct use of "cms".
@kandrosov, please let me know if if there is a real need to keep current design - if it is present I am fine with reverting it.

Reply:
@mbluj This design choice has been made before my time (for example, it is already present in the code referenced in quite old recommendations for 2017 https://github.com/greyxray/TauAnalysisTools/blob/CMSSW_9_4_X_tau-pog_RunIIFall17/TauAnalysisTools/python/runTauIdMVA.py#L29). As far as I see, it can safely be replaced by `import FWCore.ParameterSet.Config as cms`

Reply:
OK, thanks for comment and check. 
It has just been replaced.